Turn -webkit-mask-box-image into a shorthand

Loirooriol opened the pull request:
#### 3bdc3f2824cf4d01a15ee402d7de99b099e8b3ec
<pre>
Turn -webkit-mask-box-image into a shorthand
<a href="https://bugs.webkit.org/show_bug.cgi?id=249115">https://bugs.webkit.org/show_bug.cgi?id=249115</a>

Reviewed by Darin Adler.

* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-webkit-mask-box-image.html: Fix bug with omitted values.
* LayoutTests/fast/css/image-set-parsing-expected.txt: getPropertyCSSValue doesn&apos;t work for shorthands.
* LayoutTests/fast/css/image-set-parsing.html: getPropertyCSSValue doesn&apos;t work for shorthands.
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt: Shorthands are not indexed in style declarations.
* Source/WebCore/css/CSSProperties.json: Turn -webkit-mask-box-image into a shorthand.
* Source/WebCore/css/StyleProperties.cpp: Serialize -webkit-mask-box-image as a shorthand.
(WebCore::StyleProperties::getPropertyValue const):
(WebCore::StyleProperties::borderImagePropertyValue const):
* Source/WebCore/css/parser/CSSPropertyParser.cpp: Parse -webkit-mask-box-image as a shorthand.
(WebCore::CSSPropertyParser::consumeBorderImage):
(WebCore::CSSPropertyParser::parseShorthand):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp: Refactoring.
(WebCore::CSSPropertyParserHelpers::consumeReflect):
(WebCore::CSSPropertyParserHelpers::consumeWebkitBorderImage): Deleted.
* Source/WebCore/rendering/style/RenderStyle.h: Delete dead code.
(WebCore::RenderStyle::initialOutlineStyleIsAuto):
(WebCore::RenderStyle::initialNinePieceImageForMask): Deleted.
* Source/WebCore/style/StyleBuilderConverter.h: Delete dead code.
(WebCore::Style::BuilderConverter::convertBorderMask): Deleted.

Canonical link: <a href="https://commits.webkit.org/257723@main">https://commits.webkit.org/257723@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d19cda79f0a4cd65fab0358d0c65d835703b554

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99804 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8984 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32892 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109170 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169409 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103807 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9817 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86275 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92268 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107080 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105571 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7434 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90726 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34184 "Passed tests") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22111 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2800 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23623 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2746 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/46002 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5306 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8881 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43093 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4609 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
This all looks good.

Loirooriol committed:
Turn -webkit-mask-box-image into a shorthand
https://bugs.webkit.org/show_bug.cgi?id=249115

Reviewed by Darin Adler.

* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/fast/css/getComputedStyle/getComputedStyle-webkit-mask-box-image.html: Fix bug with omitted values.
* LayoutTests/fast/css/image-set-parsing-expected.txt: getPropertyCSSValue doesn't work for shorthands.
* LayoutTests/fast/css/image-set-parsing.html: getPropertyCSSValue doesn't work for shorthands.
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Shorthands are not indexed in style declarations.
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt: Shorthands are not indexed in style declarations.
* Source/WebCore/css/CSSProperties.json: Turn -webkit-mask-box-image into a shorthand.
* Source/WebCore/css/StyleProperties.cpp: Serialize -webkit-mask-box-image as a shorthand.
(WebCore::StyleProperties::getPropertyValue const):
(WebCore::StyleProperties::borderImagePropertyValue const):
* Source/WebCore/css/parser/CSSPropertyParser.cpp: Parse -webkit-mask-box-image as a shorthand.
(WebCore::CSSPropertyParser::consumeBorderImage):
(WebCore::CSSPropertyParser::parseShorthand):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp: Refactoring.
(WebCore::CSSPropertyParserHelpers::consumeReflect):
(WebCore::CSSPropertyParserHelpers::consumeWebkitBorderImage): Deleted.
* Source/WebCore/rendering/style/RenderStyle.h: Delete dead code.
(WebCore::RenderStyle::initialOutlineStyleIsAuto):
(WebCore::RenderStyle::initialNinePieceImageForMask): Deleted.
* Source/WebCore/style/StyleBuilderConverter.h: Delete dead code.
(WebCore::Style::BuilderConverter::convertBorderMask): Deleted.

Canonical link: https://commits.webkit.org/257723@main
